akka-stream/src/main/scala/akka/stream/Materializer.scala

```diff
@@ -31,6 +35,24 @@ abstract class Materializer {
    */
   implicit def executionContext: ExecutionContextExecutor
 
+  /**
+   * Interface for stages that need timer services for their functionality. Schedules a
+   * single task with the given delay.
+   *
+   * @return A [[akka.actor.Cancellable]] that allows cancelling the timer. Cancelling is best effort, if the event
+   *         has been already enqueued it will not have an effect.
+   */
+  def scheduleOnce(delay: FiniteDuration, task: Runnable): Cancellable
+
+  /**
+   * Interface for stages that need timer services for their functionality. Schedules a
+   * repeated task with the given interval between invocations.
+   *
+   * @return A [[akka.actor.Cancellable]] that allows cancelling the timer. Cancelling is best effort, if the event
+   *         has been already enqueued it will not have an effect.
+   */
+  def schedulePeriodically(interval: FiniteDuration, task: Runnable): Cancellable
+
```

Comment:
I think we started without it in the Source timer, but then someone needed it and we added it, so I was thinking we should add it immediately now.

The two timers is a good idea though for the rare cases when it is needed. I like that, since we have those named timers.